third_party/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/TMTensor/IR/TMTensorOps.td

```diff
@@ -189,6 +189,56 @@ def TMTensor_ScatterOp : TMTensor_Op<"scatter",
   }];
 }
 
+def TMTensor_SortOp : TMTensor_Op<"sort",
+    [DeclareOpInterfaceMethods<ScalarLoopOpInterface,
+        ["generateScalarImplementation"]>]> {
+  let summary = "Sort operator";
+  let description = [{
+    Based on XLA operation semantics, sorts the given `operands` at the given
```

Comment:
This is a little bit weird: is this a tm tensor one as where you are producing this you can't use StableHLO? Else why not use that StableHLO op? (If it is a case about mix of ops not being supported, then that's also something we should look at and has been on my list but not many requests).

Reply:
So, this is needed in the lowering path from `torch-mlir` -> `linalg`.
We don't use `StableHLO` in that path, so we'd be requiring to use this TMTensor op. Here's the corresponding [Torch-MLIR to TMTensor SortOp](https://github.com/llvm/torch-mlir/pull/2003) PR, that'll need the current PR to go in.

CC: @powderluv